src/stream.c

```diff
@@ -33,11 +33,17 @@ static uv_stream_t* luv_check_stream(lua_State* L, int index) {
   // we get its metatable and check that we get `true` back
   // when looking the metatable up in the "uv_stream" table.
   lua_getfield(L, LUA_REGISTRYINDEX, "uv_stream");
-  lua_getmetatable(L, index < 0 ? index - 1 : index);
+  isStream = lua_getmetatable(L, index < 0 ? index - 1 : index);
+  if (!isStream) goto fail;
   lua_rawget(L, -2);
   isStream = lua_toboolean(L, -1);
   lua_pop(L, 2);
-  if (isStream) { return handle; }
+  if (!isStream) goto fail;
+  // cast the userdata to uv_stream_t
+  handle = *(uv_stream_t**)udata;
+  if (!handle || !handle->data) goto fail;
```

Comment:
```suggestion
  if (!handle->data) goto fail;
```

Reply:
I should've commited the changes myself instead of using that Github commit suggestion thing, my bad